saleor/graphql/product/resolvers.py

```diff
@@ -134,9 +130,9 @@ def resolve_report_product_sales(period, channel_slug=None) -> ChannelQsContext:
 
     # exclude draft and canceled orders
     exclude_status = [OrderStatus.DRAFT, OrderStatus.CANCELED]
-    qs = qs.exclude(order_lines__order__status__in=exclude_status).filter(
-        order_lines__order__channel__slug=str(channel_slug)
-    )
+    qs = qs.exclude(order_lines__order__status__in=exclude_status)
+    if channel_slug:
+        qs = qs.filter(order_lines__order__channel__slug=str(channel_slug))
```

Comment:
This resolver has no option to work without `channel_slug`. 
All order_lines in this `qs` should be in the same currency.

I think that `channel_slug` should be required for `reportProductSales` query.